### PR TITLE
CI & cmake changes

### PR DESCRIPTION
**Pull request by qnfm**
Fully implement CI jobs for building under VS environment
Align the cmake test file with the VS generator.

NB: The ci change in this commit rely on my pr branch #256. I'm not sure whether should I make pr directly or just wait upstream to merged #256 then make this pr.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 include(GNUInstallDirs)
+if (CMAKE_GENERATOR MATCHES "Visual Studio")
+set(OQS_PROV_BINARY_DIR ${CMAKE_BINARY_DIR}/lib/Debug)
```
Comment (baentsch) [inline test/CMakeLists.txt:3]: Is this path always correct -- particularly if CMAKE_BUILD_TYPE is set to "Release"? Or does that variable not propagate into VS?

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -199,3 +199,109 @@ jobs:
...
+        platform:
+          - arch: win64
+            oqsconfig: -DOQS_ALGS_ENABLED=STD
+            osslconfig: no-shared no-fips VC-WIN64A
+        toolchain: 
+#        - .CMake/toolchain_windows_x86.cmake
+        - .CMake/toolchain_windows_amd64.cmake
+        msarch:
+          - x64
+# - arm64
```
Comment (baentsch) [inline .github/workflows/windows.yml:218]: Am I assuming correct that github CI doesn't support these architecture types? Worthwhile adding a comment? Or completely eliminate these 2 lines?

**Review by baentsch — COMMENTED** [review]
> NB: The ci change in this commit rely on my pr branch https://github.com/open-quantum-safe/oqs-provider/pull/256. I'm not sure whether should I make pr directly or just wait upstream to merged https://github.com/open-quantum-safe/oqs-provider/pull/256 then make this pr.

I'm not quite sure what you mean: Does #263 add to & improve the changes in #256? If so, we could simply close #256 and only merge #263, right? I was holding back #256 because of your final comment "Not final solution".

Otherwise things LGTM with two single comments where I'd appreciate feedback to properly understand things. Also waiting for CI to complete. Thanks very much for this work!

File: test/CMakeLists.txt (hunk 1 above)
Comment (qnfm) [reply test/CMakeLists.txt:3]: I set the cmake build type to Debug in ci, which seems a reasonable choice to produce more output. So yes

**Comment by qnfm** [comment]
> > NB: The ci change in this commit rely on my pr branch #256. I'm not sure whether should I make pr directly or just wait upstream to merged #256 then make this pr.
> 
> I'm not quite sure what you mean: Does #263 add to & improve the changes in #256? If so, we could simply close #256 and only merge #263, right? I was holding back #256 because of your final comment "Not final solution".
> 
> Otherwise things LGTM with two single comments where I'd appreciate feedback to properly understand things. Also waiting for CI to complete. Thanks very much for this work!

File: test/CMakeLists.txt (hunk 1 above)
Comment (baentsch) [reply test/CMakeLists.txt:3]: What happens when someone sets the build type to something different?

Comment (baentsch) [reply test/CMakeLists.txt]: Improved in https://github.com/open-quantum-safe/oqs-provider/pull/263/commits/ce60ecd5e0640f834cb0e41a0049eeb6cc9dcbee

**Review by baentsch — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**